### ISSUE TITLE
[Vai] Server side pagination - Sync lag of Kubernetes resources counts when quering the counts after resources creation 

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#46701]
**Setup**
- Rancher version: `v2.9-head` k3s HA setup AWS.
- Feature: `ui-sql-cache` enabled
- Performance: Backend Pagination enabled.

**Describe the bug**
The api resources counts from the API (i.e. api/pods) has a lag after creation. More noticeable on HA setups where the Rancher replicas have to sync the cache.

**To Reproduce**

Using pods as api resources for tests.
Get the initial total v1/pods count. `initialTotal`
Create a number of api resources in succession. 25 pods used for this. 
Query the API again without waiting long to get the new count. `initialTotal + 25`

**Result**

The new total queried to the pods count from the API doesn't show the new api resources. The new resources count in this case pods are sync after over 20 seconds.

**Expected Result**
The count returned after api resources creation shouldn't take long to show the new total.

**Additional context**
https://github.com/rancher/dashboard/pull/11587#discussion_r1716641884

`POST` api request to `/v1/pods` (done to create 25 pods serially)
Payload:

```json
{
    "metadata": {
        "namespace": "test-namespace",
        "labels": {
            "workload.user.cattle.io/workloadselector": "pod-test-namespace-t1"
        },
        "name": "t1",
        "annotations": {}
    },
    "spec": {
        "containers": [
            {
                "imagePullPolicy": "Always",
                "name": "container-0",
                "volumeMounts": [],
                "image": "nginx:alpine"
            }
        ],
        "initContainers": [],
        "imagePullSecrets": [],
        "volumes": [],
        "affinity": {}
    }
}
```

After that it's just a `GET` call to `v1/pods` to get the `body.count` and check the expected `initialTotal` + 25

**Comment by izaac** [comment]
Possibly related: https://github.com/rancher/rancher/issues/46030

**Comment by moio** [comment]
I now believe https://github.com/rancher/lasso/pull/98 has fixed this.

I failed to reproduce this on lasso and rancher current `main` brancher, see below.

I therefore expect the next builds based on master should not exhibit this problem any more (2.10 rcs at the latest, possibly also in the 2.9.x line but it's not fully decided yet).

## Instructions

- get [an API token](https://ranchermanager.docs.rancher.com/v2.9/api/api-tokens)
- open a shell to each one of 3 Rancher replica pods (I used k9s). Inside of each pod, run the following script that continuously outputs the pod count:

```sh
while true
do
  curl --insecure -s -u $API_TOKEN -X GET "https://localhost/v1/pods" | grep -oh '"count":[0-9]*'
  sleep 0.1
done
```
- from an external host with access to Rancher, use the `steve_create_pods.js` k6 scripts to create any number of pods:

```sh
k6 run -e USERNAME=$RANCHER_USERNAME -e PASSWORD=$RANCHER_PASSWORD -e BASE_URL=$RANCHER_URL -e POD_COUNT=25 ./steve_create_pods.js
```

The script is stored here: https://github.com/moio/dartboard/blob/k6_steve_create/k6/steve_create_pods.js

Result:


https://github.com/user-attachments/assets/7a313f0c-bb85-47b8-9444-5600aa58d7dd


Note that the counter goes down as the script initially cleans up pods, and then creates the new ones - very quickly (although not instantaneously) all pods get up to date with their counts.

**Comment by rancher-max** [comment]
It appears the relevant code is not yet in rancher in the required release lines. We need a lasso bump in `main` and `release/v2.9` in order to see and validate this. Ref: https://github.com/rancher/rancher/blob/main/go.mod#L135 and https://github.com/rancher/rancher/blob/release/v2.9/go.mod#L136

**Comment by moio** [comment]
@rancher-max you're right, #47329 should address that

**Comment by izaac** [comment]
This was specially noticeable with the pods resources as it was the most problematic making the tests run results inconsistent. 

The result now is an instant return of all resources and the UI navigation is really responsive.
Test done: I did a few runs of the ``pods.spec.ts` without the polling function. 


Rancher version: `v2.10-98818170477f8b4e9e866bc38c575ae0abf31490-head`
Dashboard version: `master 9f5bbdd`

`ui-sql-cache` enabled
`Server-Side Pagination` enabled.

`ui-index`  and `ui-dashboard-index` pointing to the latest location.

Closing as done.
